caddyhttp/log/log.go

```diff
@@ -66,6 +67,12 @@ func (l Logger) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error) {
 
 			// Write log entries
 			for _, e := range rule.Entries {
+				// Mask IP Address, will not modify if default
+				hostip, hostport, err := net.SplitHostPort(r.RemoteAddr)
+				if err == nil {
+					r.RemoteAddr = e.Log.MaskIP(hostip) + ":" + hostport
```

Comment:
This won't work for IPv6 addresses. `2001:feed:c0fe::` + Port 1234 should be `[2001:feed:c0fe::]:1234`. Use `net.JoinHostPort`.